octobot/community/authentication.py

```diff
@@ -364,6 +364,12 @@ async def refresh_selected_bot(self):
             await self.supabase_client.fetch_bot(self.user_account.bot_id)
         )
 
+    async def refresh_selected_bot_if_unset(self):
+        if not self.user_account.has_selected_bot_data():
+            self.user_account.set_selected_bot_raw_data(
+                await self.supabase_client.fetch_bot(self.user_account.bot_id)
```

Comment:
Should we also check that `self.user_account.bot_id` is not None ?

Reply:
it's initialized in constructor, it should never be None, we are fine :)

Reply:
Great!